### PR TITLE
[PVR][guiinfo][Estuary] New GUI info label Listitem.IsPremiere + impl…

### PR DESCRIPTION
**Pull request by ksooo**
…ementation for PVR EPG info tags + visualization in Estuary.

Requested in the forum: https://forum.kodi.tv/showthread.php?tid=345895

Guide window:
screenshot000

PVR info dialog:
screenshot002

@phunkyfish what do you think about this?

Like always, I'm not a skinner, so if somebody wants to fine tune my skin changes, you are more than welcome. @ronie maybe?

### CODE REVIEW

File: addons/skin.estuary/xml/Includes_PVR.xml [hunk 1]
```diff
@@ -461,10 +461,28 @@
...
+						<texture>$VAR[PVRTimerIcon]</texture>
+					</control>
+					<control type="image">
+						<visible>!ListItem.IsPremiere</visible>
+						<left>6</left>
+						<top>35</top>
+						<width>16</width>
+						<height>16</height>
 						<texture>$VAR[PVRTimerIcon]</texture>
 					</control>
```
Comment (ronie) [inline addons/skin.estuary/xml/Includes_PVR.xml]: you can use a single image control for the PVRTimerIcon:
```
<control type="image">
	<left>6</left>
	<top>35</top>
	<width>16</width>
	<height>16</height>
	<texture>$VAR[PVRTimerIcon]</texture>
	<animation effect="slide" start="0,0" end="19,0" condition="ListItem.IsPremiere">Conditional</animation>
</control>
```

File: addons/skin.estuary/xml/Includes_PVR.xml [hunk 2]
```diff
@@ -498,10 +516,28 @@
...
+					<control type="image">
+						<visible>!ListItem.IsPremiere</visible>
 						<left>6</left>
 						<top>35</top>
-						<width>20</width>
-						<height>20</height>
+						<width>16</width>
+						<height>16</height>
 						<texture>$VAR[PVRTimerIcon]</texture>
 					</control>
```
Comment (ronie) [inline addons/skin.estuary/xml/Includes_PVR.xml]: same here

**Review by ronie — CHANGES_REQUESTED** [review]

File: addons/skin.estuary/xml/Includes_PVR.xml (hunk 1 above)
Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: Nice. Will change that, ofc.

File: addons/skin.estuary/xml/Includes_PVR.xml (hunk 2 above)
Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: dto

File: xbmc/pvr/PVRGUIInfo.cpp [hunk 3]
```diff
@@ -1246,6 +1246,19 @@ bool CPVRGUIInfo::GetListItemAndPlayerBool(const CFileItem *item, const CGUIInfo
...
       break;
+    case LISTITEM_IS_PREMIERE:
+      if (item->IsEPG())
+      {
+        const std::shared_ptr<CPVREpgInfoTag> epgTag = item->GetEPGInfoTag();
+        const CDateTime firstAired = epgTag->FirstAiredAsUTC();
+        const CDateTime start = epgTag->StartAsUTC();
+        if (firstAired.IsValid() && start.IsValid())
+        {
+          bValue = firstAired == start;
```
Comment (phunkyfish) [inline xbmc/pvr/PVRGUIInfo.cpp]: So it first aired is only a date can this still work?

Comment (phunkyfish) [reply xbmc/pvr/PVRGUIInfo.cpp]: I think it's probably fair to say an episode is new on its airing date regardless of time.

Comment (ksooo) [reply xbmc/pvr/PVRGUIInfo.cpp]: Yep, will change that.

File: addons/skin.estuary/xml/Includes_PVR.xml (hunk 1 above)
Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: Fixed.

File: addons/skin.estuary/xml/Includes_PVR.xml (hunk 2 above)
Comment (ksooo) [reply addons/skin.estuary/xml/Includes_PVR.xml]: Fixed.

File: xbmc/pvr/PVRGUIInfo.cpp (hunk 3 above)
Comment (ksooo) [reply xbmc/pvr/PVRGUIInfo.cpp]: Fixed.

**Review by phunkyfish — APPROVED** [review]
Looks good from the backend.

**Comment by phunkyfish** [comment]
Bonus feature. Being able to search for `new` programmes!

**Comment by phunkyfish** [comment]
Could technically be part of this PR.

**Comment by ksooo** [comment]
> Bonus feature. Being able to search for new programmes!
> Could technically be part of this PR (but I really have no idea what’s involved).

 I put this on my list, but not for this PR. ;-)

**Comment by ksooo** [comment]
@ronie I do not really the look of the new 'premiere.png' - I actually reused 'rating.png'. Could you maybe come up with a png for a star with (far) more edges? Or in case you have a better idea how to visualize "new", the better...

**Comment by ksooo** [comment]
Rebased and much nicer image for the new indicator in the guide window now. Thanks @ronie for that.

**Review by ronie — APPROVED** [review]

**Comment by emveepee** [comment]
This is great addition but note that for Sports, News shows etc this often is shown as "Live" not "New"  Also any reference to Premiere should be avoided since there are other available flags like Season/Series Premiere that may or may not be "New"

Martin

**Comment by ksooo** [comment]
So, what do you suggest? Listitem.IsNew?

**Comment by phunkyfish** [comment]
If you were to use IsNew then you could have IsLive and IsPremiere in the future.

**Comment by ksooo** [comment]
Sounds good. Will prepare a PR next days.

**Comment by emveepee** [comment]
You could eventually add new EPG_TAG fields for isNew etc rather than triggering from the original air date to allow for different broadcast countries and even timezones. Something can be new here in Canada that was shown months earlier in the UK or US.

**Comment by ksooo** [comment]
Sure, and that is for another PR.

**Comment by emveepee** [comment]
That's fine, once I have Matrix build system I can work on that.  However until flags are available I think there is a logic issue comparing start time with the original air date because a date point is only valid in one timezone.  It probably should be defined as localtime not UTC. If not data from Schedules Direct and zap2xml sources in Canada and the US (by far the most used here)  most comparisons will fail.  I can convert the original air data to UTC but that then it will display the wrong information and all PVR addons would have to do this.

**Comment by ksooo** [comment]
>  I can convert the original air data to UTC but that then it will display the wrong information and all PVR addons would have to do this.

 EPG_TAG.firstAired is defined (!) as UTC time. So, every addon has to do this anyway. 

https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/xbmc_epg_types.h#L100

If something is not displayed correctly, PVR core needs to be fixed. However, implementation for the respective guiinfo label looks okay to me.

https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/PVRGUIInfo.cpp#L608

**Comment by emveepee** [comment]
Ok we never implemented original_air_date so I was unaware of the conversion and that certainly looks fine.   Defining the field as localtime would have made sense from a documentation point of view since it is kind of hidden to those of use with limited knowledge of the backend.  Perhaps the EPG_TAGS documentation should now explain how this is now used as a New indicator?

**Pull request merged**